src/View/Form/AbstractContext.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Cake\View\Form;
+
+use Cake\Network\Request;
+
+/**
+ * Provides common functionality for ContextInterface implementations.
+ */
+abstract class AbstractContext implements ContextInterface
+{
+    /**
+     * The request object.
+     *
+     * @var \Cake\Network\Request
+     */
+    protected $_request;
+
+    /**
+     * The request type of the attached form.
+     *
+     * @var string
+     */
+    protected $_requestType;
+
+    /**
+     * Constructor.
+     *
+     * @param \Cake\Network\Request $request The request object.
+     */
+    protected function __construct(Request $request)
+    {
+        $this->_request = $request;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setRequestType($requestType)
+    {
+        $this->_requestType = $requestType;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function val($field)
+    {
+        if ($this->_requestType === 'get') {
+            return $this->_request->query($field);
+        }
+        return $this->_request->data($field);
+    }
```

Comment:
I'm not sure this class has enough meat to justify its existence. A tiny bit of duplication would be less complex from my perspective.